applicationset/generators/generator_spec_processor_test.go

```diff
@@ -73,7 +73,82 @@ func TestMatchValues(t *testing.T) {
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "set",
 				},
-				Spec: argov1alpha1.ApplicationSetSpec{},
+				Spec: argov1alpha1.ApplicationSetSpec{
+					GoTemplate: false,
+				},
+			}
+
+			results, err := Transform(argov1alpha1.ApplicationSetGenerator{
+				Selector: testCase.selector,
+				List: &argov1alpha1.ListGenerator{
+					Elements: testCase.elements,
+					Template: emptyTemplate(),
+				}},
+				data,
+				emptyTemplate(),
+				&applicationSetInfo, nil)
+
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, testCase.expected, results[0].Params)
+		})
+	}
+}
+
+func TestMatchValuesGoTemplate(t *testing.T) {
```

Comment:
Would you mind adding a test for list values, just so it's documented somewhere? Probably an edge case, but good to cover.

Reply:
Could you elaborate on this? Do you mean tests for values that slice types?

Reply:
Yep, exactly. So if my generator returns `{"list": ["1", "2", "3"]}`, can I select on the second item in `list`? If so, what's the selector look like?

Reply:
So I've just done a quick test with this. For go templates It works fine but it seems without the go templates the tests fail as there's an expectation that the values will always be of type `map[string]interface{}` and then subsequently that the value of that interface is a string. I suspect this may be a greater piece of work to add that support in to the older style templates.

Reply:
Ah interesting. No sweat then, I think it's a far-edge case.

Reply:
Okay, I've added a test case for the go templates but omitted the test for argo templates.